plugins/modules/helm_cli.py

```diff
@@ -0,0 +1,593 @@
+
+    if not HAS_YAML:
+        module.fail_json(msg="Could not import the yaml python module. Please install `yaml` package.")
+    changed = False
+
+    bin_path = module.params.get('binary_path')
+    chart_ref = module.params.get('chart_ref')
+    chart_repo_url = module.params.get('chart_repo_url')
+    chart_repo_username = module.params.get('chart_repo_username')
+    chart_repo_password = module.params.get('chart_repo_password')
+    chart_version = module.params.get('chart_version')
+    release_name = module.params.get('release_name')
+    release_namespace = module.params.get('release_namespace')
+    release_state = module.params.get('release_state')
+    release_values = module.params.get('release_values')
+    tiller_host = module.params.get('tiller_host')
+    tiller_namespace = module.params.get('tiller_namespace')
+    update_repo_cache = module.params.get('update_repo_cache')
+
+    # Helm options
+    disable_hook = module.params.get('disable_hook')
+    force = module.params.get('force')
+    kube_context = module.params.get('kube_context')
+    kubeconfig_path = module.params.get('kubeconfig_path')
+    purge = module.params.get('purge')
+    wait = module.params.get('wait')
+    wait_timeout = module.params.get('wait_timeout')
+
+    if bin_path is not None:
+        helm_cmd_common = bin_path
+    else:
+        helm_cmd_common = module.get_bin_path('helm', required=True)
+
+    is_helm_2 = get_helm_client_version(helm_cmd_common)
+
+    # Helm 2 need tiller, Helm 3 and higher doesn't
+    if is_helm_2:
+        if tiller_host is not None:
+            helm_cmd_common += " --host=" + tiller_namespace
+        else:
+            helm_cmd_common += " --tiller-namespace=" + tiller_namespace
+
+    if kube_context is not None:
+        helm_cmd_common += " --kube-context " + kube_context
+
+    if kubeconfig_path is not None:
+        helm_cmd_common += " --kubeconfig " + kubeconfig_path
+
+    if update_repo_cache:
+        run_repo_update(helm_cmd_common)
+
+    # Get real/deployed release status
+    release_status = get_release_status(helm_cmd_common, release_name, release_namespace)
+
+    # keep helm_cmd_common for get_release_status in module_exit_json
+    helm_cmd = helm_cmd_common
+    if release_state == "absent" and release_status is not None:
+        if is_helm_2:
+            helm_cmd = delete_2(helm_cmd, release_name, purge, disable_hook)
+        else:
+            helm_cmd = delete_3(helm_cmd, release_name, release_namespace, purge, disable_hook)
+        changed = True
+    elif release_state == "present":
+
+        if chart_version is not None:
+            helm_cmd += " --version=" + chart_version
+
+        if chart_repo_url is not None:
+            helm_cmd += " --repo=" + chart_repo_url
+            if chart_repo_username is not None and chart_repo_password is not None:
+                helm_cmd += " --username=" + chart_repo_username
+                helm_cmd += " --password=" + chart_repo_password
```

Comment:
Is there any other way to specify `password` rather than the current implementation?

Reply:
It looks like it may be possible to use the `helm registry login` subcommand to pipe the credentials in instead of adding them directly, though I'm not super familiar with how it works https://helm.sh/docs/topics/registries/#login

Reply:
`helm registry login` is available only for helm 3
`username` and `password` are used only for basic auth on the chart repository.
A alternative solution will be to make a `helm repo add` with the username and password before calling the module.
Helm will store these credentials inside its config files and will inject it automatically when the module will run commands.

Maybe the better solution is to remove these 2 arguments and put in the doc to run a `helm repo add` before ?